awscli/customizations/s3/utils.py

```diff
@@ -517,3 +517,134 @@ def __new__(cls, message, error=False, total_parts=None, warning=None):
 class IOCloseRequest(_IOCloseRequest):
     def __new__(cls, filename, desired_mtime=None):
         return super(IOCloseRequest, cls).__new__(cls, filename, desired_mtime)
+
+
+class RequestParamsMapper(object):
+    """A utility class that maps CLI params to request params
+
+    Each method in the class maps to a particular operation and will set
+    the request parameters depending on the operation and CLI parameters
+    provided.
+    """
+    @classmethod
+    def map_put_object_params(cls, request_params, cli_params):
```

Comment:
A docstring (or a class docstring because it looks like they all have similar signatures) that shows a basic example as well as documents the params would be helpful.